src/inmanta/agent/executor.py

```diff
@@ -244,36 +247,153 @@ class ResourceInstallSpec:
+        cleanup
     """
 
-    def __init__(self, env_path: str, threadpool: ThreadPoolExecutor):
+    def __init__(self, env_path: str, threadpool: Optional[ThreadPoolExecutor]):
         super().__init__(env_path=env_path)
         self.thread_pool = threadpool
+        self.inmanta_venv_status_file = pathlib.Path(self.env_path) / const.INMANTA_VENV_STATUS_FILENAME
+        self.folder_name = pathlib.Path(self.env_path).name
 
-    async def create_and_install_environment(self, blueprint: EnvBlueprint) -> None:
+    def create_and_install_environment(self, blueprint: EnvBlueprint) -> None:
         """
         Creates and configures the virtual environment according to the provided blueprint.
 
         :param blueprint: An instance of EnvBlueprint containing the configuration for
             the pip installation and the requirements to install.
         """
-        loop = asyncio.get_running_loop()
         req: list[str] = list(blueprint.requirements)
         self.init_env()
         if len(req):  # install_for_config expects at least 1 requirement or a path to install
-            install_for_config = functools.partial(
-                self.install_for_config,
+            self.install_for_config(
                 requirements=list(pkg_resources.parse_requirements(req)),
                 config=blueprint.pip_config,
             )
-            await loop.run_in_executor(self.thread_pool, install_for_config)
+
+    def is_correctly_initialized(self) -> bool:
+        """
+        Was the venv correctly initialized: the inmanta status file exists
+        """
+        return self.inmanta_venv_status_file.exists()
+
+    def touch_status_file(self) -> None:
+        """
+        Touch the inmanta status file
+        """
+        self.inmanta_venv_status_file.touch()
+
+    def last_used(self) -> datetime.datetime:
+        """
+        Retrieve the last modified timestamp of the inmanta status file
+        """
+        assert self.is_correctly_initialized()
+        return datetime.datetime.fromtimestamp(self.inmanta_venv_status_file.stat().st_mtime).astimezone()
+
+    async def clean(self) -> None:
+        """
+        Remove the venv of the executor through the thread pool.
+        This method is supposed to be used by the VirtualEnvironmentManager with the lock associated to this executor!
+        """
+        loop = asyncio.get_running_loop()
+
+        if self.is_correctly_initialized():
+            LOGGER.debug(
+                "Removing venv %s because it was inactive for %d s, which is longer then the retention " "time of %d s.",
+                self.env_path,
+                self.get_idle_time().total_seconds(),
+                cfg.executor_venv_retention_time.get(),
+            )
+        else:
+            LOGGER.debug(
+                "Removing venv %s because it is incomplete (broken venv),.",
+                self.env_path,
+            )
+        await loop.run_in_executor(self.thread_pool, self.remove_venv)
```

Comment:
[It is not an issue](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor), right? 

If the `venv` is not present in the `self._environment_map`, then we don't know which thread pool this `ExecutorVirtualEnvironment` can use. Therefore, we must rely on the default one.

Reply:
What I said is not entirely correct because we know that the thread pool is set by the `MPManager`.
![image](https://github.com/user-attachments/assets/151aba11-9034-4624-8f56-b57bd9e47abf)